src/realm/query_expression.cpp

```diff
@@ -510,10 +510,11 @@ void LinkCount::evaluate(size_t index, ValueBase& destination)
         auto sz = links.size();
 
         if (sz == 0) {
-            destination.init(true, 0);
+            destination = Value<int64_t>(0);
```

Comment:
The difference is having no results to match on, versus having one result of value 0 to match with.

Reply:
I deliberately made this distinction. I guess we have never really discussed what the correct result is. My position would be that objects where the collection in question does not even exist, should not be matched.